### PR TITLE
GraphQL: Allow specifying Args on GraphQLFieldResolver (optional)

### PR DESCRIPTION
**Pull request by tgriesser**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

File: types/graphql/type/definition.d.ts [hunk 1]
```diff
@@ -262,9 +262,9 @@ export type GraphQLIsTypeOfFn<TSource, TContext> = (
     info: GraphQLResolveInfo
 ) => boolean | Promise<boolean>;
 
-export type GraphQLFieldResolver<TSource, TContext> = (
+export type GraphQLFieldResolver<TSource, TContext, TArgs = { [argName: string]: any }> = (
```
Comment (IvanGoncharov) [inline types/graphql/type/definition.d.ts]: @tgriesser In `graphql-js` it described as following 
```js
export type GraphQLFieldResolver<TSource, TContext> = (
```
https://github.com/graphql/graphql-js/blob/master/src/type/definition.js#L717

I strongly believe that TS typings should be in sync with official Flow typings.
Please consider contributing your changes to `graphql-js` first.

**Review by IvanGoncharov — CHANGES_REQUESTED** [review]

File: types/graphql/type/definition.d.ts (hunk 1 above)
Comment (tgriesser) [reply types/graphql/type/definition.d.ts]: Ah cool will do

Comment (tgriesser) [reply types/graphql/type/definition.d.ts]: @IvanGoncharov Related PR: https://github.com/graphql/graphql-js/pull/1197

**Comment by tgriesser** [comment]
@IvanGoncharov I think this should be good to go now that `graphql-js` is updated, let me know if you need any changes.

**Review by IvanGoncharov — APPROVED** [review]

**Comment by corydeppen** [comment]
@tgriesser I continue to be unable to include args in a resolver function without a TS error. Do you have an example of a resolver function taking an arg that doesn't throw a TS error using `"noImplicitAny": "true"` and `"strictFunctionTypes": true` (or `"strict": true`) with TS >= 2.6?

**Pull request merged**